### PR TITLE
Show "(guest)" by guest users' names, when setting enabled

### PR DESCRIPTION
**Pull request by chrisbobbe**
Stacked on my current revision for #5806.

I've put "(guest)" in italics where it's rendered with React components, since it was possible and convenient to do it there.

### DM compose content input placeholder text before / after:

| Before | After |
| --- | --- |
| image | image |

### Other "after" screenshots:

image
image
image
image
image
image
image
image

Fixes: #5804

### CODE REVIEW

**Comment by chrisbobbe** [comment]
Here's a before / after for this commit. With, in both, an uncommitted patch to declare the `user_list_incomplete` client capability (see #5808):

CustomProfileFields: Use styled UserItem for (unknown user)

| Before | After |
| --- | --- |
| image | image |

**Comment by chrisbobbe** [comment]
Oh, and a whole-PR before-and-after showing the changes to `AvatarItem`, i.e., the avatar with an overlaid slashed-circle "remove" button, that shows you which users you've selected in the new-group-DM form. In particular this shows the effect of this commit:

AvatarItem: Stop carelessly amputating parts of people's names

| Before | After |
| --- | --- |
| image | image |

**Comment by chrisbobbe** [comment]
Thanks for the review! Revision pushed.

**Comment by chrisbobbe** [comment]
Oops! Pushed to the wrong PR branch; back now.

**Review by gnprice — COMMENTED** [review]
Thanks @chrisbobbe for building this! I agree with the strategy of starting by centralizing how we compute the name text. This looks to be quite thorough.

Several comments below, all of them small.

File: src/api/modelTypes.js [hunk 1]
```diff
@@ -311,8 +303,7 @@ export type User = {|
   // TODO(server-3.0): Replaced in FL 1 by bot_owner_id
   +bot_owner?: string,
 
-  // TODO(server-4.0): New in FL 59
-  +role?: Role,
+  +role: Role,
```
[the remark this answers is not in the document]
Comment (chrisbobbe) [reply src/api/modelTypes.js:306]: Ah OK, makes sense. Thanks for the feedback!

File: src/compose/ComposeBox.js [hunk 2]
```diff
@@ -713,7 +714,14 @@ const ComposeBox: React$AbstractComponent<Props, ImperativeHandle> = forwardRef(
...
 
-  const placeholder = getComposeInputPlaceholder(narrow, ownUserId, allUsersById, streamsById);
+  const placeholder = getComposeInputPlaceholder(
+    narrow,
+    ownUserId,
+    allUsersById,
+    streamsById,
+    enableGuestUserIndicator,
+    _,
+  );
```
[the remark this answers is not in the document]
Comment (chrisbobbe) [reply src/compose/ComposeBox.js:724]: Yeah.

**Comment by chrisbobbe** [comment]
Thanks for the review! Revision pushed, and I fixed the failing tests. I also added a TODO in a commit at the end for a place we don't yet have "(guest)" indicators: @-mentions in message content. Hmm.

**Comment by gnprice** [comment]
Thanks! All LGTM now, except there are some rebase conflicts to resolve.

**Comment by chrisbobbe** [comment]
Thanks! Revision pushed, resolving those conflicts.

File: src/storage/__tests__/migrations-test.js [hunk 3]
```diff
@@ -102,13 +102,18 @@ describe('migrations', () => {
     },
   };
 
-  // What `base` becomes after all migrations.
-  const endBase = {
+  // What `base` becomes after migrations up through 62.
+  const base62 = {
     ...base52,
-    migrations: { version: 62 },
```
Comment (gnprice) [inline src/storage/__tests__/migrations-test.js:108]: This `migrations` line should stay — it makes the description of this "base" value accurate, and it's what we do with the other "baseN" values.

**Review by gnprice — COMMENTED** [review]
Thanks! Looks good except one small glitch below. Please go ahead and merge after that.

**Comment by chrisbobbe** [comment]
Thanks! Done.

**Pull request merged**